Mathlib/Algebra/TrivSqZeroExt.lean

```diff
@@ -226,6 +226,20 @@ instance module [Semiring S] [AddCommMonoid R] [AddCommMonoid M] [Module S R] [M
     Module S (tsze R M) :=
   Prod.instModule
 
+/-- The trivial square-zero extension is nontrivial if it is over a nontrivial ring. -/
+instance instNontrivial_of_left {R M : Type*} [Nontrivial R] [Nonempty M] :
+    Nontrivial (TrivSqZeroExt R M) := by
+  inhabit M
+  have : Zero M := ⟨default⟩
```

Comment:
I hadn't thought about this. Can you get there in fewer lines by using surjectivity of fst rather than objectively of inl?

Reply:
The proof of that surjectivity will require an `inhabit` also